### PR TITLE
Aligned the templates with Kubernetes

### LINKED ISSUE #11: Use a proper Cassandra docker image
For the moment, we are using the Cassandra docker image under my own namespace. We should decide which image to use: Docker Hub's? Our own, based on GCE's (kubernetes official example)? Another one?

### PR DESCRIPTION
**Pull request by jpkrohling**
Fixes #11

### CODE REVIEW

**Comment by jpkrohling** [comment]
This PR is mainly aligning the OpenShift templates with Kubernetes.

File: all-in-one/jaeger-all-in-one-template.yml [hunk 1]
```diff
@@ -40,29 +40,26 @@ metadata:
     name: jaeger-infra
     jaeger-infra: jaeger-template-all-in-one
 objects:
-- apiVersion: v1
-  kind: DeploymentConfig
+- apiVersion: extensions/v1beta1
```
Comment (pavolloffay) [inline all-in-one/jaeger-all-in-one-template.yml]: What does this effectively mean? A newer OC version?

Comment (jpkrohling) [inline all-in-one/jaeger-all-in-one-template.yml]: A relatively new version of `oc`, yes. This should be OK, as we require a similar version on the Kubernetes template.

Comment (pavolloffay) [inline all-in-one/jaeger-all-in-one-template.yml]: For curiosity can I ask what version? Should we document this?

**Comment by pavolloffay** [comment]
@Jeeva @kevinearls could you please test this? both prod and all-in-one (this also with mvn verify -Pe2e).

File: all-in-one/jaeger-all-in-one-template.yml (hunk 1 above)
Comment (jpkrohling) [inline all-in-one/jaeger-all-in-one-template.yml]: Good question. I *think* this was new in 1.5, but I'm not 100% sure. This is the feature in question: https://kubernetes.io/docs/concepts/workloads/controllers/deployment/

**Comment by kevinearls** [comment]
@pavolloffay Sure, I'll try it now

File: all-in-one/jaeger-all-in-one-template.yml (hunk 1 above)
Comment (pavolloffay) [inline all-in-one/jaeger-all-in-one-template.yml]: Maybe here https://github.com/openshift/origin/releases/tag/v1.4.0-alpha.1

Comment (jpkrohling) [inline all-in-one/jaeger-all-in-one-template.yml]: Perhaps. In any case, I'd not mention any versions here, as it has the potential of becoming a rabbit hole. Once this gets really supported, then we should certainly determine the minimum versions.

**Comment by kevinearls** [comment]
@pavolloffay So far I can get the all-in-one to deploy correctly, but not the production template.  Unfortunately I don't have time to analyze the failures at the moment.  @jkandasa can you take a look at this?  Otherwise I can try again first thing tomorrow.

**Comment by kevinearls** [comment]
@jpkrohling @pavolloffay I can get the production template to deploy, and can get to the UI, but so far I haven't been able to run the tests.  My minishift instance is behaving strangely, possibly because I updated Fedora this morning.  I have now updated to minishift 1.3 but still haven't be able to straighten everything out.

I'll have to defer to @jkandasa to check the rest of this.

**Comment by pavolloffay** [comment]
@kevinearls you can also use `oc cluster up`

**Comment by jpkrohling** [comment]
> Change jaeger service name in tests to `jaeger-query`.

I'll fix it on this PR, but it has not been caused by the changes from this PR.

**Comment by pavolloffay** [comment]
@jpkrohling thanks for fixing it

**Pull request merged**